content/contributing/opencue/Getting started/getting-started-linux.md

```diff
@@ -0,0 +1,312 @@
+
+1. Verify that the output window doesn’t show any errors.
+
+   If it does, double-check that you have set up the database correctly,
+   including permissions, and have set the Program arguments correctly.
+
+## Create a virtual environment
+
+OpenCue consists of multiple Python components which are interrelated. These components
+include RQD, CueGUI, and the Python API.
+
+We recommend creating a Python virtual environment specifically for development use that you can
+use for all components. This will help keep dependencies synchronized across your OpenCue
+deployment.
+
+1. Open a terminal and change to the root folder of your OpenCue Git clone.
+
+1. Create a virtual environment named `venv-dev`:
+
+   ```shell
+   sudo python3 -m venv venv-dev
+   ```
+   
+   Your virtual environment can be named whatever you want; this rest of this guide assumes
+   you're using one named `venv-dev`.
+   
+1. Activate the virtual environment:
+
+   ```shell
+   source venv-dev/bin/activate
+   ```
+
+## Configure PyCharm
+
+Similar to the virtual environment, we recommend configuring your IDE as a single project
+containing all of the Python components. PyCharm is used here.
+
+1. Open PyCharm and choose **Open**. Select the root folder of the git repository.
+
+1. Open the PyCharm preferences and navigate to **Project: opencue > Project Interpreter**.
+
+1. Add a new interpreter, using the following settings:
+   
+   - **Virtual environment**
+   - **Existing environment**
+   - **Interpreter** set to `<path to git repository>/venv-dev/bin/python`
+
+1. PyCharm should prompt you to install 'Package requirements'. This will install OpenCue's
+   Python dependencies into your virtual environment.
+   
+   This can take a few minutes, namely to download `PySide2`.
+
+1. In order for inter-dependencies within the code to work in PyCharm you need to mark
+   each components as a source directory. In the PyCharm file browser, right-click on
+   each OpenCue component and click on **Mark Directory As > Sources Root**. You
+   need to do this for:
+   
+   - `cueadmin/`
+   - `cuegui/`
+   - `cuesubmit/`
+   - `proto/`
+   - `pycue/`
+   - `pyoutline/`
+   - `rqd/`
+
+## Generate gRPC Python code
+
+The OpenCue data model and gRPC API are defined using
+[`.proto` Protobuf files](https://developers.google.com/protocol-buffers). To make use of
+these definitions at runtime the Protobuf files must first be "compiled" into native code —
+Java and Python in our case, though Protobuf supports many languages. This increases consistency
+across OpenCue components and reduces code repetition.
+
+Generating the Python versions of OpenCue's `.proto` files for RQD and the PyCue library is
+currently a manual process. To generate the Python code:
+
+1. Open a terminal, change to the root directory of the Git clone, and activate your
+   virtual environment as described above.
+
+1. Install the required gRPC tools:
+
+    ```shell
+    sudo pip -m install grpcio-tools
```

Comment:
`sudo` also shouldn't be necessary here, if you're using your virtual environment.